Assets/Scripts/Managers/NetworkPlayerMappingManager.cs

```diff
@@ -0,0 +1,119 @@
+using System.Collections;
+using System.Linq;
+using System.Collections.Generic;
+using UnityEngine;
+using Photon.Pun;
+using Photon.Realtime;
+using UtilityExtensions;
+
+public class NetworkPlayerMappingManager : MonoBehaviourPunCallbacks, IConnectionCallbacks
+{
+    public static NetworkPlayerMappingManager instance;
+    List<int> freePlayers;
+    Dictionary<int, int> playerOwners = new Dictionary<int, int>();
+    PhotonView photonView;
+
+    void Awake()
+    {
+        if (instance == null)
+        {
+            freePlayers = new List<int> { 1, 2, 3, 4 };
+            instance = this;
+        }
+        else
+        {
+            Destroy(this);
+        }
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        Utility.Print("Mapping manager", LogLevel.Error);
+        photonView = GetComponent<PhotonView>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+
+    }
+
+    public override void OnJoinedRoom()
+    {
+        Utility.Print("Joined room", LogLevel.Error);
+        this.FrameDelayCall(() => OwnNextPlayer(PhotonNetwork.LocalPlayer.ActorNumber));
```

Comment:
why frame delay call? I feel like frame delay calls will cause issues and we should start phasing it out, is there a particular reason you used it here?

Reply:
I agree on phasing it out. I'm just not sure how to handle without it here. It seems like when I don't have it, the buffered RPC sometimes isn't sent before this is called so the state is all fucked.